pagopa/io-app
# chore: [IOPID-1181] Add email uniqueness validation FF

## ⚠️ Depends on https://github.com/pagopa/io-services-metadata/pull/745 ⚠️ ##

## Short description
This PR adds the new email uniqueness validation remote FF and the corresponding selector.

## List of modifications
* [package.json](https://github.com/pagopa/io-app/blob/e67d73e8fcef33df4046895d9c7f7bb08c6c1f25/package.json): update metadata definitions.
* [ts/components/helpers/withValidatedEmail.tsx](https://github.com/pagopa/io-app/blob/e67d73e8fcef33df4046895d9c7f7bb08c6c1f25/ts%2Fcomponents%2Fhelpers%2FwithValidatedEmail.tsx): remove unused code.
* [ts/features/fastLogin/store/selectors/__tests__/emailUniquenessValidation.test.ts](https://github.com/pagopa/io-app/blob/e67d73e8fcef33df4046895d9c7f7bb08c6c1f25/ts%2Ffeatures%2FfastLogin%2Fstore%2Fselectors%2F__tests__%2FemailUniquenessValidation.test.ts): add FF selector tests.
* [ts/features/fastLogin/store/selectors/index.ts](https://github.com/pagopa/io-app/blob/e67d73e8fcef33df4046895d9c7f7bb08c6c1f25/ts%2Ffeatures%2FfastLogin%2Fstore%2Fselectors%2Findex.ts): add FF selector.
* [ts/navigation/OnboardingNavigator.tsx](https://github.com/pagopa/io-app/blob/e67d73e8fcef33df4046895d9c7f7bb08c6c1f25/ts%2Fnavigation%2FOnboardingNavigator.tsx):
* [ts/navigation/ProfileNavigator.tsx](https://github.com/pagopa/io-app/blob/e67d73e8fcef33df4046895d9c7f7bb08c6c1f25/ts%2Fnavigation%2FProfileNavigator.tsx):
* [ts/sagas/startup/__tests__/checkProfileEmailSaga.test.tsx](https://github.com/pagopa/io-app/blob/e67d73e8fcef33df4046895d9c7f7bb08c6c1f25/ts%2Fsagas%2Fstartup%2F__tests__%2FcheckProfileEmailSaga.test.tsx):
* [ts/sagas/startup/checkAcknowledgedEmailSaga.ts](https://github.com/pagopa/io-app/blob/e67d73e8fcef33df4046895d9c7f7bb08c6c1f25/ts%2Fsagas%2Fstartup%2FcheckAcknowledgedEmailSaga.ts):
* [ts/sagas/startup/checkEmailSaga.ts](https://github.com/pagopa/io-app/blob/e67d73e8fcef33df4046895d9c7f7bb08c6c1f25/ts%2Fsagas%2Fstartup%2FcheckEmailSaga.ts):
* [ts/screens/profile/CduEmailInsertScreen.tsx](https://github.com/pagopa/io-app/blob/e67d73e8fcef33df4046895d9c7f7bb08c6c1f25/ts%2Fscreens%2Fprofile%2FCduEmailInsertScreen.tsx):
* [ts/screens/profile/ProfileDataScreen.tsx](https://github.com/pagopa/io-app/blob/e67d73e8fcef33df4046895d9c7f7bb08c6c1f25/ts%2Fscreens%2Fprofile%2FProfileDataScreen.tsx): replace the local FF with the new FF selector.

## How to test
1. Enable local FF:
```
CDU_NEW_FLOW=YES
```
2. Enable remote flag on the [dev server](https://github.com/pagopa/io-dev-api-server/blob/00c956a2176dcf667f9334a539295d6c523a90ca/src/payloads/backend.ts#L112):
```
emailUniquenessValidation: {
      min_app_version: {
        ios: "1.0.0",
        android: "1.0.0"
      }
    },
```
3. With the remote flag enabled you should see the new email uniqueness validation flow. Without the flag you should see the standard email validation flow:

<details open><summary>Details</summary>
<video src="https://github.com/pagopa/io-app/assets/16268789/d6f3e18f-ef5a-4a7e-8daa-951d605dfebe" />
<p>

</p>
</details>
